Sources/SmithyTestUtil/JSONComparator.swift

```diff
@@ -47,6 +47,8 @@ fileprivate func anyValuesAreEqual(_ lhs: Any?, _ rhs: Any?) -> Bool {
         return anyDictsAreEqual(lhsDict, rhsDict)
     } else if let lhsArray = lhs as? [Any], let rhsArray = rhs as? [Any] {
         return anyArraysAreEqual(lhsArray, rhsArray)
+    } else if let lhn = lhs as? NSNumber, let rhn = rhs as? NSNumber {
+        return lhn == rhn
```

Comment:
Handles case where JSON values that are logically equivalent (e.g., `1` and `1.0`) but are different concrete specific types